Xamarin.PropertyEditing.Windows/TextBoxEx.cs

```diff
@@ -116,12 +116,14 @@ private void FocusSelect()
 				SelectAll ();
 		}
 
+		internal bool BubbleAfterReturn { get; set; } = false;
+
 		private void OnPreviewKeyDown (object sender, System.Windows.Input.KeyEventArgs e)
 		{
 			if (e.Key == Key.Enter) {
 				var expression = GetBindingExpression (TextProperty);
 				expression?.UpdateSource ();
-				e.Handled = true;
+				if (!BubbleAfterReturn) e.Handled = true;
```

Comment:
We can actually probably just remove the handled setter altogether, I don't think it's needed.